src/DGmethods/test/Sphere/advection_sphere.jl

```diff
@@ -0,0 +1,290 @@
+using CLIMA.MPIStateArrays
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+
+const uid, vid, wid = 1:3
+const integration_testing = true
+const γ_exact = 7 // 5
+
+if !@isdefined integration_testing
+  const integration_testing =
+    parse(Bool, lowercase(get(ENV,"JULIA_CLIMA_INTEGRATION_TESTING","false")))
+  using Random
+end
+
+# preflux computation: NOT needed for this test
+@inline function preflux(Q, _...)
+  γ::eltype(Q) = γ_exact
+  @inbounds ρ = Q[1]
+    ρinv = 1 / ρ
+    u, v, w, P = 0, 0, 0, 0
+    (P, u, v, w, ρinv)
+end
+
+#{{{ advectionflux
+advectionflux!(F, Q, QV, aux, t) = advectionflux!(F, Q, QV, aux, t, preflux(Q)...)
+@inline function advectionflux!(F,Q,QV,aux,t,P,u,v,w,ρinv)
+    @inbounds begin
+        u,v,w = aux[uid], aux[vid], aux[wid]
+        ρ=Q[1]
+        F[1,1], F[2,1], F[3,1]=u*ρ, v*ρ, w*ρ
+    end
+end
+#}}} advectionflux
+
+#{{{ wavespeed
+@inline function wavespeed(n, Q, vel, t...)
+    abs(dot(vel,n))
+end
+#}}} wavespeed
+
+#{{{ Cartesian->Spherical
+@inline function cartesian_to_spherical(x,y,z,format)
+    #Conversion Constants
+    if format == "degrees"
+        c=180/π
+    elseif format == "radians"
+        c=1.0
+    end
+    λ_max=2π*c
+    λ_min=0*c
+    ϕ_max=+0.5*π*c
+    ϕ_min=-0.5*π*c
+
+    #Conversion functions
+    r = hypot(x,y,z)
+    λ=atan(y,x)*c
+    ϕ=asin(z/r)*c
+    return (r, λ, ϕ)
+end
+#}}} Cartesian->Spherical
+
+#{{{ velocity initial condition
+@inline function velocity_init!(vel, x, y, z)
+    @inbounds begin
+        DFloat = eltype(vel)
+        (r, λ, ϕ) = cartesian_to_spherical(x,y,z,"radians")
+        #w = 2 * DFloat(π) * cos(ϕ) #Case 1 -> shear flow
+        w = 2 * DFloat(π) * cos(ϕ) * r #Case 2 -> solid body flow
+        uλ, uϕ = w, 0
+        vel[uid] = -uλ*sin(λ) - uϕ*cos(λ)*sin(ϕ)
+        vel[vid] = +uλ*cos(λ) - uϕ*sin(λ)*sin(ϕ)
+        vel[wid] = +uϕ*cos(ϕ)
+    end
+end
+#}}} velocity initial condition
+
+# initial condition
+function advection_sphere!(Q, t, x, y, z, vel)
+    DFloat = eltype(Q)
+    rc=1.5
+    (r, λ, ϕ) = cartesian_to_spherical(x,y,z,"radians")
+    if (abs(r-rc) <= 0.25)
+        Q[1] = exp(-((3λ)^2 + (3ϕ)^2))
+    else
+        #            Q[1]=0
+        Q[1] = exp(-((3λ)^2 + (3ϕ)^2))
+    end
+end
+
+#{{{ Main
+function main(mpicomm, DFloat, topl, N, timeend, ArrayType, dt)
+    grid = DiscontinuousSpectralElementGrid(topl,
+                                            FloatType = DFloat,
+                                            DeviceArray = ArrayType,
+                                            polynomialorder = N,
+                                            meshwarp = Topologies.cubedshellwarp)
+
+    #{{{ numerical fluxex
+    numflux!(x...) = NumericalFluxes.rusanov!(x..., advectionflux!, wavespeed,preflux)
+    numbcflux!(x...) =  F.=0 #zero flux at the boundaries: not entirely accurate but good enough
```

Comment:
`F` is not an argument for this function as written. Should be 'numbcflux!(F, _...) = F.=0` I think.